plugins/woocommerce/src/Blocks/BlockTypes/BlockifiedProductDetails.php

```diff
@@ -263,13 +301,111 @@ private function has_accordion( $parsed_block ) {
 	}
 
 	/**
-	 * Get the frontend script handle for this block type.
+	 * Validate hooked blocks data. Remove duplicated entries with the same title
+	 * and invalid entries with invalid content. Log errors to the WC logger.
 	 *
-	 * @see $this->register_block_type()
-	 * @param string $key Data to get, or default to everything.
-	 * @return array|string|null
+	 * @param array $hooked_blocks { Hooked blocks data.
+	 *   @type string $title Title of the hooked block.
+	 *   @type string $content Content of the hooked block, as block markup.
+	 * }
+	 *
+	 * @return array Validated hooked blocks.
 	 */
-	protected function get_block_type_script( $key = null ) {
-		return null;
+	private function validate_hooked_blocks( $hooked_blocks ) {
+		$logger                  = wc_get_logger();
+		$validated_hooked_blocks = [];
+
+		foreach ( $hooked_blocks as $block ) {
+			$invalid = ! is_array( $block ) ||
+			! isset( $block['title'] ) ||
+			! isset( $block['content'] ) ||
+			! is_string( $block['title'] ) ||
+			! is_string( $block['content'] );
```

Comment:
Nitpick, I think these need to be indented a bit more.

```suggestion
					! isset( $block['title'] ) ||
					! isset( $block['content'] ) ||
					! is_string( $block['title'] ) ||
					! is_string( $block['content'] );
```